build_tools/circle/build_doc.sh

```diff
@@ -58,6 +58,25 @@ get_build_type() {
         return
     fi
     changed_examples=$(echo "$filenames" | grep -E "^examples/(.*/)*plot_")
+    if [ -n "$filenames" ]
+    then
+        for af in ${filenames[@]}
+        do
+            if [ $af != "build_tools/circle/build_doc.sh" ]
+            then
+                page_figure=$(grep figure $af | grep auto_example | awk -F "/" '{print $NF}' | sed 's/sphx_glr_//' | awk -F "_" '{OFS="_";$NF=""; print $0}')
```

Comment:
I think this is missing the `examples` root directory.

Please see comment regarding a chained bash script.

Reply:
You are right : the `example` directory is not taken into account.
But the `example` directory is irrelevant to this line of code. As said by @adrinjalali I'm looking for examples in modified `.rst` pages in order to build them even if they have not been modified.
It looks to me that examples in that directory are never included in the `.rst` pages because they are independent pages. Am I wrong?

Reply:
Sorry.. I didn't understand your comment, my answer is nonsense... you mean in the script path right?
Well I tested the build process and apparently the `EXAMPLES_PATTERN` argument accepts regexp, so no need for the complete path...

Reply:
@cmarmo ‘s code is I feel more readable

Reply:
Yes, I forgot that `EXAMPLES_PATTERN` is a regex, so the current solution works.

If @cmarmo 's version is more readable, we should go with the current version. I would prefer slightly to only include `.rst` in `filenames` to restrict reduce the items in the loop.